lib/mongo/collection.ex

```diff
@@ -560,106 +560,85 @@ defmodule Mongo.Collection do
 
     load_function =
       quote unquote: false do
-        attribute_names = @attributes |> Enum.map(&elem(&1, 0))
+        attribute_names = Enum.map(@attributes, fn {name, opts} -> {name, opts[:name]} end)
 
         embed_ones =
           @embed_ones
           |> Enum.filter(fn {_name, mod, _opts} -> Collection.has_load_function?(mod) end)
-          |> Enum.map(fn {name, mod, _opts} -> {name, mod} end)
+          |> Enum.map(fn {name, mod, opts} -> {name, {opts[:name], mod}} end)
 
         embed_manys =
           @embed_manys
           |> Enum.filter(fn {_name, mod, _opts} -> Collection.has_load_function?(mod) end)
-          |> Enum.map(fn {name, mod, _opts} -> {name, mod} end)
+          |> Enum.map(fn {name, mod, opts} -> {name, {opts[:name], mod}} end)
 
         def load(map, use_atoms \\ false)
 
-        def load(nil, _use_atoms) do
-          nil
-        end
+        def load(nil, _use_atoms), do: nil
 
         def load(xs, use_atoms) when is_list(xs) do
           Enum.map(xs, fn map -> load(map, use_atoms) end)
         end
 
-        def load(map, false) when is_map(map) do
-          struct =
-            Enum.reduce(
-              unquote(attribute_names),
-              %__MODULE__{},
-              fn name, result ->
-                Map.put(result, name, map[Atom.to_string(name)])
-              end
-            )
+        def load(map, use_atoms) when is_map(map) do
+          struct = Enum.reduce(unquote(attribute_names), %__MODULE__{}, fn {name, src_name}, result -> Map.put(result, name, map[src_name(src_name, use_atoms)]) end)
```

Comment:
We should remember that a load and a dump are used very often. Maybe we can make it faster. I created a little benchee test: 

```elixir
defmodule MyColl do
  use Mongo.Collection
  collection "cards" do
    attribute :title_a, String.t(), default: "new title", name: :a
    attribute :title_b, String.t(), default: "new title", name: :b
    attribute :title_c, String.t(), default: "new title", name: :c
    attribute :title_d, String.t(), default: "new title", name: :d
    attribute :title_e, String.t(), default: "new title", name: :e
    attribute :title_f, String.t(), default: "new title", name: :f
    attribute :title_g, String.t(), default: "new title", name: :g
  end

  def load_2(doc) do
    attribute_names = [{:title_a, "a"}, {:title_b, "b"}, {:title_c, "c"}, {:title_d, "d"}, {:title_e, "e"}, {:title_f, "f"}, {:title_g, "g"}]
    Enum.reduce(attribute_names, %__MODULE__{}, fn {name, src_name}, result -> Map.put(result, name, doc[src_name]) end)
  end
end

doc = %{"a" => "test a", "b" => "test b", "c" => "test c", "d" => "test d", "e" => "test e", "f" => "test f", "g" => "test g"}

Benchee.run(
  %{
    "load" => fn -> MyColl.load(doc) end,
    "load-2" => fn -> MyColl.load_2(doc) end,
  }
)
```

Here the results:

```elixir
Benchmark suite executing with the following configuration:
warmup: 2 s
time: 5 s
memory time: 0 ns
parallel: 1
inputs: none specified
Estimated total run time: 14 s

Benchmarking load...
Benchmarking load-2...

Name             ips        average  deviation         median         99th %
load-2        1.53 M      652.95 ns  ±2753.45%         583 ns         750 ns
load          1.12 M      890.66 ns  ±2302.37%         833 ns        1000 ns

Comparison:
load-2        1.53 M
load          1.12 M - 1.36x slower +237.71 ns
```
Therefore we should move the work for `map[src_name(src_name, use_atoms)]` outside of the function into the macro itself because we do this work again and again.